BreakingChanges.md

```diff
@@ -5,6 +5,7 @@
 * Obsolete api is removed
 * CompatArg is marked as obsolete
 * Nullability is enabled for public api for .NET 8+. Workaround: you can disable nullability `<Nullable>disable</Nullable>` on project if you don't want to use it. more info https://learn.microsoft.com/en-us/dotnet/csharp/nullable-references
+* Move some core type into NSubstitute.Internal namespace
```

Comment:
maybe we should note something that the internal namespace is not meant to be used.